docs/book/src/cli-usage.md

```diff
@@ -0,0 +1,256 @@
+# CLI
+### Usage
+The Topiary CLI uses a number of subcommands to delineate functionality.
+These can be listed with `topiary --help`; each subcommand then has its
+own, dedicated help text.
+
+<!-- DO NOT REMOVE THE "usage" COMMENTS -->
+<!-- usage:start:ROOT -->
+```
+CLI app for Topiary, the universal code formatter.
+
+Usage: topiary [OPTIONS] <COMMAND>
+
+Commands:
+  format      Format inputs
+  visualise   Visualise the input's Tree-sitter parse tree
+  config      Print the current configuration
+  prefetch    Prefetch all languages in the configuration
+  completion  Generate shell completion script
+  help        Print this message or the help of the given subcommand(s)
+
+Options:
+  -C, --configuration <CONFIGURATION>  Configuration file [env: TOPIARY_CONFIG_FILE]
+  -v, --verbose...                     Logging verbosity (increased per occurrence)
+  -h, --help                           Print help
+  -V, --version                        Print version
+```
+<!-- usage:end:ROOT -->
+
+#### Format
+
+<!-- DO NOT REMOVE THE "usage" COMMENTS -->
+<!-- usage:start:format -->
+```
+Format inputs
+
+Usage: topiary format [OPTIONS] <--language <LANGUAGE>|FILES>
+
+Arguments:
+  [FILES]...
+          Input files and directories (omit to read from stdin)
+
+          Language detection and query selection is automatic, mapped from file extensions defined
+          in the Topiary configuration.
+
+Options:
+  -t, --tolerate-parsing-errors
+          Consume as much as possible in the presence of parsing errors
+
+  -s, --skip-idempotence
+          Do not check that formatting twice gives the same output
+
+  -l, --language <LANGUAGE>
+          Topiary language identifier (for formatting stdin)
```

Comment:
I'm wondering whether the book or the help message should indicate how to learn the list of valid language identifiers.